src/semidiscretization/semidiscretization.jl

```diff
@@ -61,6 +61,20 @@ for regression tests.
 calc_error_norms(u_ode, t, analyzer, semi::AbstractSemidiscretization, cache_analysis) = calc_error_norms(cons2cons, u_ode, t, analyzer, semi, cache_analysis)
 
 
+# An internal wrapper of a semidiscretization and a right-hand side function.
+# This allows setting up an `ODEProblem` "knowing" the semidiscretization
+# without passing it as an explicit parameter. Doing so enables the use of
+# more precompiled methods from OrdinaryDiffEq.jl and can thus reduce latency.
+struct RHSWrapper{Semi, Rhs!}
+  semi::Semi
+  rhs!::Rhs!
+end
```

Comment:
IIRC, you mentioned that you wanted to deliberately lose type information such that we would not have to recompile as much. Or are these two different potential improvements that I am mixing up?

Reply:
I didn't want to lose the type information here - but reducing the number of types may improve latency, so it may be worth checking in other cases later.

Reply:
OK, thanks for the clarification!